ls1intum/Artemis
# Refactoring/programming exercise/refactor programming exercise assessment components into module

### Checklist

### Motivation and Context



Preparation step for https://github.com/ls1intum/Artemis/issues/940 (tutor dashboard for programming exercises).
Fixes issue: https://github.com/ls1intum/Artemis/issues/950.

### Description


- Refactor the manual result and repo export components into a new module and make them reusable.
- Remove programming exercise specific functionality from the generic services (e.g. exerciseService should not have a method for exporting repos)
- Some smaller refactorings for loading states
- Added option to manual result dialog to only download the repository of specific students (necessary for the submission presented in the tutor dashboard)
- Show grading instructions + reduce whitespace from https://github.com/ls1intum/Artemis/issues/950. Note: I did not remove the normal form element whitespace because it is needed for the form validation (errors will look crunched otherwise)

:warning: Even though there are still traces in the code, it is not possible to update the manual result (we always create a new one). Not sure if this was removed on purpose, if not we could restore the functionality in a follow up PR.

:information_source: Because we don't have much time left for the integration of the tutor dashboard, we have to save time by neglecting the tests (as discussed with @krusche). In the case of this PR there is not much new code anyway, it is mostly rearrangement of existing code.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration > Programming Exercise with manual assessment enabled > Scores
3. Download Repos / New Result should still work as expected

### Screenshots



**Result dialog:**

![image](https://user-images.githubusercontent.com/28230611/67096759-1714e580-f1b9-11e9-84c2-d51a2bedbde8.png)

**Single Repository Export:**

![image](https://user-images.githubusercontent.com/28230611/67086593-69e2a300-f1a1-11e9-8771-ec6b39e5056f.png)
